### PR TITLE
Fix: postRender automatically setting blank component as completed

### LINKED ISSUE #45: Blank components should always be marked as complete
### Subject of the issue/enhancement/features
To avoid course completion prevention when blank components are used, they should be marked as complete upon postRender on the page

### Your environment
* FW 5.X
* Chrome
* All devices

### Steps to reproduce
Create a course with blank components and page level progress

### Expected behaviour
PLP should fill immediately on page load. Clicking PLP will show blank components individually have completed

### Actual behaviour
Course will never complete as Blank components do not get marked this way

### PR DESCRIPTION
**Pull request by cahirodoherty-learningpool**
[//]: # (Please title your PR according to eslint commit conventions)
[//]: # (See https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-eslint#eslint-convention for details)

[//]: # (Link the PR to the original issue)
Fixes #45 

[//]: # (Delete Fix, Update, New and/or Breaking sections as appropriate)
### Fix
* postRender automatically setting blank component as completed on page view

[//]: # (List appropriate steps for testing if needed)
### Testing
1. Create a course with blank components and page level progress
2. PLP should fill immediately on page load
3. Clicking PLP will show blank components individually have completed

### CODE REVIEW

**Comment by paulstevendev** [comment]
Our requirements are that the blank component behaves according to the "Is this optional" setting. We use this component to determine if a user has viewed a page by placing a blank component at the top of each page with all other components set to optional. Therefore if you place a blank component at the top of the page and set it to NOT optional I would expect the menu of the course not to detect the page as complete until the user opens the page. This is how it currently works with the open source blank component but the Learning Pool one appears to act as optional despite the setting that has been set for "Is this optional"

**Comment by cahirodoherty-learningpool** [comment]
Yes this is part of the difference that you are seeing @paulstevendev. The LearningPool AT will override these components and set them as optional always. We want to remove this when the new component functionality is added.
The changes to the blank component would effectively make page completion occur upon opening when a blank component is on that page and is the only non-optional component. Does that make sense?

**Comment by paulstevendev** [comment]
@cahirodoherty-learningpool - the suggestion @oliverfoster has made sounds like a good compromise. If you did have a page where all the non blank components are set to optional then adding any blank components with a default setting of optional would result in page completion occur upon opening the course. However forcing the blank component to always act as optional is not an ideal solution for us so hopefully you can go with the suggestion by @oliverfoster

**Comment by cahirodoherty-learningpool** [comment]
Yeah that works for us. I've just pushed another commit

**Comment by paulstevendev** [comment]
Thanks Cahir

**Review by chris-steele — APPROVED** [review]
This makes eminent sense

File: properties.schema [hunk 1]
```diff
@@ -10,6 +10,11 @@
       "enum": ["full-width", "half-width", "both"],
       "default": "both",
       "editorOnly": true
-    }
+    },
+    "_isOptional": {
+      "type": "boolean",
+      "required": false,
+      "default": false
```
Comment (oliverfoster) [inline properties.schema:17]:
```suggestion
      "default": true
```

File: js/blankView.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ export default class BlankView extends ComponentView {
 
   postRender() {
     this.setReadyStatus();
-    this.setupInviewCompletion();
+    this.setCompletionStatus();
```
Comment (oliverfoster) [inline js/blankView.js:11]:
```suggestion
    this.setupInviewCompletion();
```

**Review by oliverfoster — CHANGES_REQUESTED** [review]

File: js/blankView.js (hunk 2 above)
Comment (oliverfoster) [reply js/blankView.js:11]: Can we leave this as it was, as the `_isOptional: false` + inview can be useful to auto complete on page end?

**Comment by oliverfoster** [comment]
Taking @paulstevendev as implicit +1

**Comment by paulstevendev** [comment]
It doesn't appear that there are any properties appearing in the AT now for the blank component so it is impossible to set this to NOT optional.

**Comment by cahirodoherty-learningpool** [comment]
> It doesn't appear that there are any properties appearing in the AT now for the blank component so it is impossible to set this to NOT optional.
> 
> Testing: info [24 Jan 23 09:56:13 +00:00] Adapt authoring tool 0.11.1. info [24 Jan 23 09:56:13 +00:00] Adapt framework 5.24.8.
> 
> Blank component: 4.1.4
> 
> image

Issue to resolve this raised here: https://github.com/adaptlearning/adapt-contrib-blank/issues/51

**Pull request merged**